numpy/core/src/multiarray/calculation.c

```diff
@@ -87,9 +87,15 @@ _PyArray_ArgMinMaxCommon(PyArrayObject *op,
         op = ap;
     }
 
-    /* Will get native-byte order contiguous copy. */
-    ap = (PyArrayObject *)PyArray_ContiguousFromAny((PyObject *)op,
-                                  PyArray_DESCR(op)->type_num, 1, 0);
+    // Will get native-byte order contiguous copy.
+    // Separate handling for user-defined types.
+    if (PyArray_DESCR(op)->type_num > 0) {
```

Comment:
structured dtypes don't need it right now, but I do think calling `ensure_canonical` and not branching may be nicer?  The full fix would be to have something more sorting specific I guess.

Also why not `PyArray_FromArray`?

Reply:
> Also why not `PyArray_FromArray`?
`PyArray_ContiguousFromAny` calls `PyArray_FromAny`. I wanted to match the current implementation as closely as possible. This seems unnecessary because the argument to `_PyArray_ArgMinMaxCommon` can only be a `PyArrayObject *`, not any `PyObject *`. So it seems like we can do this with no cost - I'll change this now.